pinot-core/src/test/java/org/apache/pinot/core/common/datablock/DataBlockTest.java

```diff
@@ -93,6 +94,12 @@ public void testRowDataBlockCompatibleWithDataTableV4()
             + " from DataBlock: [" + rowFromBlock[rowId] + "], from DataTable: [" + rowFromDataTable[colId] + "]");
       }
     }
+
+    for (int colId = 0; colId < dataSchema.getColumnNames().length; colId++) {
```

Comment:
Can we add a test for the following scenarios  -- `all nulls, all non-nulls, mix of nulls and non-nulls`.

Reply:
and also cases -- nulls and non nulls on start and end of the buffer

Reply:
good idea. let me make the percentile configurable so that it can be either all nulls or all non-nulls tested